### PR TITLE
add non-MITM proxy support

### PR DESCRIPTION
**Pull request by soltysh**
This builds on top of #284 but fixes wrong yaml import which was causing the error pointed out by Standa and modifies the configmap merging. 

/assign @deads2k @stlaz

### CODE REVIEW

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -536,3 +561,18 @@ func (c *TargetConfigController) namespaceEventHandler() cache.ResourceEventHand
 		},
 	}
 }
+
+func proxyMapToEnvVars(proxyConfig map[string]string) []corev1.EnvVar {
```
Comment (mfojtik) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go]: TODO: move this to library-go :-)

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 2]
```diff
@@ -291,6 +302,20 @@ func managePod(configMapsGetter corev1client.ConfigMapsGetter, secretsGetter cor
...
+	if err := yaml.Unmarshal(operatorSpec.ObservedConfig.Raw, &observedConfig); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal the observedConfig: %v", err)
+	}
+	proxyConfig, _, err := unstructured.NestedStringMap(observedConfig, "targetconfigcontroller", "proxy")
+	if err != nil {
+		return nil, false, fmt.Errorf("couldn't get the proxy config from observedConfig: %v", err)
+	}
+
+	proxyEnvVars := proxyMapToEnvVars(proxyConfig)
+	for i, container := range required.Spec.Containers {
```
Comment (mfojtik) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go]: how about initContainers?

Comment (soltysh) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go]: We can add it to 4.3

**Pull request merged**